### PR TITLE
Support appProtocol

### LINKED ISSUE #3485: Support appProtocol
**Is your feature request related to a problem? Please describe.**
we want to use a gke's gateway for the allocator, so we want to support appProtocol.

**Additional context**

We are using grpc protocol to connect to allocator.
gke's gateway does not support `networking.gke.io/app-protocol`.
https://cloud.google.com/kubernetes-engine/docs/how-to/secure-gateway

Therefore, to connect grpc with gke gateway, you need to set `appProtocol: HTTP2`

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #3485

**Special notes for your reviewer**:

### CODE REVIEW

File: install/helm/agones/templates/service/allocation.yaml [hunk 1]
```diff
@@ -47,6 +47,9 @@ spec:
     - port: {{ .Values.agones.allocator.service.grpc.port }}
       name: {{ .Values.agones.allocator.service.grpc.portName }}
       targetPort: {{ .Values.agones.allocator.service.grpc.targetPort }}
+      {{- if .Values.agones.allocator.service.grpc.appProtocol }}
+      appProtocol: {{.Values.agones.allocator.service.grpc.appProtocol}}
```
Comment (markmandel) [inline install/helm/agones/templates/service/allocation.yaml:51]: This is the correct approach - but we'll also need a `.Values.agones.allocator.service.http.appProtocol` in the port section above (line 37 onwards).

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 2]
```diff
@@ -233,6 +233,7 @@ The following tables lists the configurable parameters of the Agones chart and t
...
 | `agones.allocator.service.serviceType`                   | The [Service Type][service] of the HTTP Service                                                                                                                                                                                                                                                                  | `LoadBalancer`                            |
+| `agones.allocator.service.grpc.appProtocol`                 | If the ServiceType is set to "appProtocol",  this is useful to connect gRPC with GKE gateway                                                                                                                                                                                                    | `""`                             |
```
Comment (markmandel) [inline site/content/en/docs/Installation/Install Agones/helm.md:236]: We'll need to make a copy of the full table and do `feature` shortcodes like we normally do.

Maybe documentation should read:

> The `appProtocol` to set on the Service for the gRPC allocation port. If left blank, no value is set.

**Review by markmandel — COMMENTED** [review]
Approach looks good - now just some cleanup 👍🏻

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 2 above)
Comment (Kalaiselvi84) [reply site/content/en/docs/Installation/Install Agones/helm.md]: The changes are present in the feature shortcodes and the description has been updated as per your suggestion.

Comment (Kalaiselvi84) [reply site/content/en/docs/Installation/Install Agones/helm.md]: Manual testing for the new changes can be found here - https://gist.github.com/Kalaiselvi84/cebc2fc16a40d3c5faf8750e0e539429

**Review by markmandel — COMMENTED** [review]
This looks good to me - @hangyoun since you reported the issue, does this look good to you?

**Comment by hangyoun** [comment]
Thank you, it will be just what I was looking for.

**Pull request merged**